### ISSUE TITLE
Are the complex templates read from local.toml and global.toml overwritten by default?

### ISSUE THREAD

**Issue by ulic-youthlic** [SuperCuber/dotter#174]
### Environment
If applicable, provide the following details:
- **OS**: Windows11 23H2
- **Dotter version**: 0.13.1

### The question
If I have a part of the complex template configuration in global.toml that is public and I just want to change the target of this template, can i write all the configuration in global.toml and just the following in the local.toml?
```toml
[my_package.files.my_template]
target = "path/to/local/file"
```
------
i had a try, but seems the new blank template in local.toml will cover the global one, and result is a blank template.

**Comment by SuperCuber** [comment]
Could you also send the `global.toml` file? To see what you're trying to override.
If you have other properties for the complex template in the global.toml you will need to re-specify them in the local.toml
The assumption you don't does make sense but it doesn't work that way currently :(

**Comment by ulic-youthlic** [comment]
Thanks for the reply,
I already know what you mean. As for why I did that, I can give you an example: I need to append a line to the profile to make my configuration take effect, so I used a complex. i wrote this template inside global.toml. However, the location of the profile is different on different machines, so I need to override the template's target attribute, which I put in local.toml. This is my usage scenario. If you ask me why I don't just add the profile to the repo, I'll tell you that PowerShell config needs some of the local part as well.
thx.

**Comment by vic-stan** [comment]
Probably related to this, i also had the question:

what's the point of having platform-specific configuration (which i think all of us want) *if* the official wiki reccomends git ignoring them? 

Do I just have to re-type my windows-specific and unix-specific paths? I get that we have templates and they also have pre-built variables that have to do with OS, but what i wanna template the `global.toml` as well, for example, based on platform?

I assume this, from the docs is relevant:
https://github.com/SuperCuber/dotter/wiki/3.-Packages-and-%60local.toml%60#included-files-and-configuration-merging

But i can't really pinpoint, in which file should this go:

> And then, on Windows you could include windows.toml like this:
> includes = [".dotter/windows.toml"]
> packages = ["vim"]

Edit: oh okay, in `local.toml` im dumb

**Comment by vic-stan** [comment]
> Probably related to this, i also had the question:
> 
> what's the point of having platform-specific configuration (which i think all of us want) _if_ the official wiki reccomends git ignoring them?
> 
> Do I just have to re-type my windows-specific and unix-specific paths? I get that we have templates and they also have pre-built variables that have to do with OS, but what i wanna template the `global.toml` as well, for example, based on platform?
> 
> I assume this, from the docs is relevant: https://github.com/SuperCuber/dotter/wiki/3.-Packages-and-%60local.toml%60#included-files-and-configuration-merging
> 
> But i can't really pinpoint, in which file should this go:
> 
> > And then, on Windows you could include windows.toml like this:
> > includes = [".dotter/windows.toml"]
> > packages = ["vim"]
> 
> Edit: oh okay, in `local.toml` im dumb

tl;dr I just wanna have platform-specific configs out of the box without typing in a `local.toml` ofr each manully, but merely cloning my repo and running a command.
I'm thinking of using conditional config file inclusions?

https://github.com/SuperCuber/dotter/wiki/4.-Complex-Target#conditional-file

for that, and have just one global, and some packages get overidden confitionally but idk if it actually helps??

**Comment by SuperCuber** [comment]
@vic-stan 
You can have a prepared `windows-local.toml` that Dotter doesn't read directly, and then `cp .dotter/windows-local.toml .dotter/local.toml` before deploying 🤷 

It will only save you the two lines of `includes` and `packages` in local.toml though so I would just use includes as is
